beamer/state_machine.py

```diff
@@ -351,13 +351,31 @@ def _handle_request_resolved(event: RequestResolved, context: Context) -> Handle
     if request is not None:
         try:
             request.l1_resolve(event.filler, event.fill_id)
+
+            discard_fill_ids = list()
+            for invalid_fill_id in request.invalid_fill_ids:
+                if event.fill_id == invalid_fill_id[0]:
+                    discard_fill_ids.append(invalid_fill_id)
+
+            for invalid_fill_id in discard_fill_ids:
+                request.invalid_fill_ids.discard(invalid_fill_id)
+
```

Comment:
This could be more concise:
```python
request.invalid_fill_ids.difference_update(value for value in request.invalid_fill_ids 
                                                            if value[0] == event.fill_id)
```